### PR TITLE
chore(eslint): enforce max-params: 2 and refactor to options-object

### PR DESCRIPTION
**Pull request by gagik**
# enforce max-params: 2 and refactor to options-object

## What

Add a `max-params: 2` ESLint rule (source files; relaxed in `.test.ts`) and refactor every function/method/constructor with >2 params to the `func({...})` options-object form across the monorepo.

## Why

The high-arity positional signatures are error-prone and hard to read. `func({...})` makes call sites self-documenting and lets each argument be added/removed by name.

## Public API changes (deliberate)

This is an intentional breaking change to the following **public/exported** consumers-facing symbols — they now take a single options object instead of positional args:

- `atlas-api-client`: `ApiClientError.fromError`
- `setup`: `AITool.updateConfig`, `AITool.getOpenConfigCommand` (via `AI_TOOL_REGISTRY`)
- `tools-atlas`: `ensureCurrentIpInAccessList`
- `tools-mongodb`: `getConnectionStringInfo`, `getIndexCheckErrorMessage`
- Constructors: `MongoDBBasedResultStorage`, `TuiSessionBase`, `ClaudeTuiSession`, `CodexTuiSession`, `TokenManager`

No compatibility wrappers are added; these call sites are migrated in-repo. (They are not part of the api-extractor-managed report surface.)

## Notes

- Cross-file call sites (including in `.test.ts`) were updated repo-wide.
- Edge cases handled: optional params and defaults preserved (`${a?: T}`, `{ a = 5 }`), parameter-property constructors get explicit `this.x = x` assignments, generics kept on the function.
- `eslint-disable-next-line max-params` used only where the positional contract is fixed by an external caller/library and cannot be options-objectized: express middleware `(req,res,next)`, Node stream `transform(chunk, encoding, callback)`, http `(req, socket, head)`, oauth `clientAuth`, braintrust `reportEval`, and function-type/interface method signatures.
- `MCP_SERVER_LIBRARY.md` and the v3 migration guide don't reference any changed symbol; the api-extractor tracked reports are unaffected.

### CODE REVIEW

File: eslint.config.js [hunk 1]
```diff
@@ -60,6 +60,7 @@ export default defineConfig([
             eqeqeq: "error",
             "no-self-compare": "error",
             "no-unassigned-vars": "error",
+            "max-params": ["error", 2],
```
[the remark this answers is not in the document]
Comment (gagik) [reply eslint.config.js:63]: Good catch to check. ESLint's core `max-params` rule does not flag interface/type method declarations (only function/method implementations), so these interface contracts aren't lint errors. The concrete implementations that ARE flagged (e.g. the `ISessionStore`/storage class methods) already carry `// eslint-disable-next-line max-params`. Verified: `eslint` on both files reports 0 errors.

File: packages/eval-tests/src/lib/seeding.ts [hunk 2]
```diff
@@ -21,14 +21,21 @@ const DEFAULT_INDEX_READY_INTERVAL_MS = 1_000;
...
-    intervalMs = DEFAULT_INDEX_READY_INTERVAL_MS
-): Promise<void> {
+    intervalMs = DEFAULT_INDEX_READY_INTERVAL_MS,
+}: {
+    client: MongoClient;
+    db: string;
+    collection: string;
+    indexNames: string[];
+    timeoutMs?: typeof DEFAULT_INDEX_READY_TIMEOUT_MS;
+    intervalMs?: typeof DEFAULT_INDEX_READY_INTERVAL_MS;
```
[the remark this answers is not in the document]
Comment (gagik) [reply packages/eval-tests/src/lib/seeding.ts]: Fixed — retyped these to `number` in commit f3abe42e4 (they were incorrectly narrowed to `typeof <constant>` literal types). Preserves the original `number` contract.

File: packages/atlas-api-client/src/apiClientError.ts [hunk 3]
```diff
@@ -17,23 +17,27 @@ export class ApiClientError extends Error {
...
-    static fromError(
-        response: Response,
-        error?: ApiError | string | Error,
-        message: string = `error calling Atlas API`
-    ): ApiClientError {
+    static fromError({
+        response,
+        error,
+        message = `error calling Atlas API`,
+    }: {
```
[the remark this answers is not in the document]
Comment (gagik) [reply packages/atlas-api-client/src/apiClientError.ts:27]: Acknowledged. This is a deliberate, intentional breaking change (per direction, no compatibility wrapper is added). It's documented in the PR description under "Public API changes (deliberate)". All in-repo call sites are migrated; the symbol isn't part of the api-extractor-managed report surface.

File: packages/setup/src/aiTool.ts [hunk 4]
```diff
@@ -204,7 +217,17 @@ export abstract class AITool {
         };
     }
 
-    updateConfig(configPath: string, env: Record<string, string>, isReadOnly: boolean, keychain: Keychain): void {
+    updateConfig({
+        configPath,
+        env,
+        isReadOnly,
+        keychain,
+    }: {
```
[the remark this answers is not in the document]
Comment (gagik) [reply packages/setup/src/aiTool.ts:225]: Acknowledged. This is a deliberate, intentional breaking change (per direction, no compatibility wrapper is added). It's documented in the PR description under "Public API changes (deliberate)". All in-repo call sites are migrated; the symbol isn't part of the api-extractor-managed report surface.

File: packages/tools-atlas/src/helpers/accessListUtils.ts [hunk 5]
```diff
@@ -76,11 +80,15 @@ export type EnsureCurrentIpResult = "added" | "already-present" | "skipped" | "f
...
-export async function ensureCurrentIpInAccessList(
-    apiClient: ApiClient,
-    projectId: string,
-    context?: ApiClientRequestContext
-): Promise<EnsureCurrentIpResult> {
+export async function ensureCurrentIpInAccessList({
+    apiClient,
+    projectId,
+    context,
+}: {
```
[the remark this answers is not in the document]
Comment (gagik) [reply packages/tools-atlas/src/helpers/accessListUtils.ts:87]: Acknowledged. This is a deliberate, intentional breaking change (per direction, no compatibility wrapper is added). It's documented in the PR description under "Public API changes (deliberate)". All in-repo call sites are migrated; the symbol isn't part of the api-extractor-managed report surface.

File: packages/tools-mongodb/src/common/connectionInfo.ts [hunk 6]
```diff
@@ -40,11 +40,15 @@ export interface ConnectionStringInfo {
...
-export function getConnectionStringInfo(
-    connectionString: string,
-    connectionInfo: ConnectionInfo,
-    hostType?: ConnectionStringHostType
-): ConnectionStringInfo {
+export function getConnectionStringInfo({
+    connectionString,
+    connectionInfo,
+    hostType,
+}: {
```
[the remark this answers is not in the document]
Comment (gagik) [reply packages/tools-mongodb/src/common/connectionInfo.ts:47]: Acknowledged. This is a deliberate, intentional breaking change (per direction, no compatibility wrapper is added). It's documented in the PR description under "Public API changes (deliberate)". All in-repo call sites are migrated; the symbol isn't part of the api-extractor-managed report surface.

File: packages/tools-mongodb/src/helpers/indexCheck.ts [hunk 7]
```diff
@@ -49,7 +49,15 @@ export function usesIndex(explainResult: Document): boolean {
 /**
  * Generate an error message for index check failure
  */
-export function getIndexCheckErrorMessage(database: string, collection: string, operation: string): string {
+export function getIndexCheckErrorMessage({
+    database,
+    collection,
+    operation,
+}: {
```
[the remark this answers is not in the document]
Comment (gagik) [reply packages/tools-mongodb/src/helpers/indexCheck.ts:56]: Acknowledged. This is a deliberate, intentional breaking change (per direction, no compatibility wrapper is added). It's documented in the PR description under "Public API changes (deliberate)". All in-repo call sites are migrated; the symbol isn't part of the api-extractor-managed report surface.

**Review by lerouxb — APPROVED** [review]
Seems fine if this is what we want as a blanket rule.
